onnx/checker.h

```diff
@@ -9,6 +9,19 @@ namespace checker {
 class ValidationError : public std::runtime_error {
  public:
   using std::runtime_error::runtime_error;
+  const char* what() const noexcept override {
+    if (!expanded_message_.empty()) {
+      return expanded_message_.c_str();
+    }
+    return std::runtime_error::what();
+  }
+  void AppendContext(const std::string& context) {
+    expanded_message_ = onnx::MakeString(
+        std::runtime_error::what(), "\n\n==> Context: ", context);
```

Comment:
the original message is not lost - we append it here. The trickery is because what() needs to returns char* and thus we should store the entire message together.

Reply:
Ah sorry didn't carefully look into this function :-)